packages/lib/date-ranges.ts

```diff
@@ -10,25 +10,46 @@ export type DateRange = {
 export type DateOverride = Pick<Availability, "date" | "startTime" | "endTime">;
 export type WorkingHours = Pick<Availability, "days" | "startTime" | "endTime">;
 
+type TravelSchedule = { startDate: Dayjs; endDate?: Dayjs; timeZone: string };
+
+function getAdjustedTimezone(date: Dayjs, timeZone: string, travelSchedules: TravelSchedule[]) {
+  let adjustedTimezone = timeZone;
+
+  travelSchedules.forEach((travelSchedule) => {
```

Comment:
Instead of looping through all the `travelSchedules` can we use a `.find()` or a `for ... of` loop and break once we find the schedule we're looking for